api-metastore/src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -398,28 +398,36 @@ public void delete(final String eventTypeName) throws EventTypeDeletionException
 
     private Multimap<TopicRepository, String> deleteEventTypeWithSubscriptions(final String eventType) {
         try {
-            return transactionTemplate.execute(action -> {
-                SubscriptionTokenLister.ListResult listResult = subscriptionTokenLister.listSubscriptions(
-                        ImmutableSet.of(eventType), Optional.empty(), Optional.empty(), null, 100);
-                while (null != listResult) {
-                    listResult.getItems().forEach(s -> {
-                        try {
-                            subscriptionRepository.deleteSubscription(s.getId());
-                        } catch (final NoSuchSubscriptionException e) {
-                            // should not happen as we are inside transaction
-                            throw new InconsistentStateException("Subscription to be deleted is not found", e);
+            return eventTypeRepository.lockingTable(EventTypeRepository.TableLock.SHARE, transactionTemplate,
+                    action -> {
+                        SubscriptionTokenLister.ListResult listResult = subscriptionTokenLister.listSubscriptions(
+                                ImmutableSet.of(eventType), Optional.empty(), Optional.empty(), null, 100);
```

Comment:
```suggestion
                        SubscriptionTokenLister.ListResult listResult = getNextResult(eventType, null);
```

Assuming we change the 2nd parater of `getNextResult()` to whatever type `listResult.getNext()` has.

Reply:
The types are the same but passing null in 2nd parameter will result in null pointer exception.

Reply:
Maybe let's remove `getNextResult()` altogether — I don't see it making code clearer or shorter for now.